ember_debug/models/profile-manager.js

```diff
@@ -5,6 +5,74 @@ const {
   run: { later, scheduleOnce },
 } = Ember;
 
+function getUnfilteredRoots(first, last, closest) {
+  const roots = [];
+  let start = null;
+  let end = null;
+  for (let i = 0; i < closest.length; i++) {
+    if (closest.item(i) === first.node) start = i;
+    else if (closest.item(i) === last.node) end = i;
+  }
+
+  if (start === null || end === null) return [];
+
+  for (let i = start; i <= end; i++) roots.push(closest.item(i));
+
+  return roots;
+}
+
+function _findRoots({ first, last, parent }) {
+  const closest = parent.childNodes;
+  if (first.node === last.node) return [first.node];
+
+  const roots = getUnfilteredRoots(first, last, closest);
+
+  return roots.filter((el) => el.nodeType === 1);
+}
+
+function makeHighlight() {
+  const node = document.createElement('div');
+  node.setAttribute('role', 'presentation');
+  node.setAttribute('class', 'ember-inspector-render-highlight');
+  return node;
+}
+function _insertHTML() {
+  return document.body.appendChild(makeHighlight());
+}
+
+function _insertStylesheet() {
+  const content = `
+    .ember-inspector-render-highlight {
+      border: 1px solid red;
+    }
+  `;
+  const style = document.createElement('style');
+  style.appendChild(document.createTextNode(content));
+  document.head.appendChild(style);
+  return style;
+}
+
+function _renderHighlight(node) {
+  if (!node?.getBoundingClientRect) {
+    return;
+  }
+  const rect = node.getBoundingClientRect();
+  const highlight = _insertHTML();
+  const { top, left, width, height } = rect;
+  const { scrollX, scrollY } = window;
+  const { style } = highlight;
+  if (style) {
+    style.position = 'absolute';
+    style.top = `${top + scrollY}px`;
+    style.left = `${left + scrollX}px`;
+    style.width = `${width}px`;
+    style.height = `${height}px`;
+    style.zIndex = `1000000`;
```

Comment:
could we have `z-index` and `position` values belongs to `ember-inspector-render-highlight` class definition? (it will be same for every node we have)

Reply:
updated.